### DISCUSSION TITLE
v2.13.0

### DISCUSSION THREAD

**Discussion by satorg** [typelevel/cats#4690] [Releases]
## Merged

All PRs merged after the v2.12.0 release.

### Contributions

All PRs created by contributors.

<details>

  - #4303 by @joroKr21
  - #4340 by @kubukoz
  - #4528 by @ahoy196
  - #4608 by @cybersaurus
  - #4612 by @isomarcte
  - #4613 by @performantdata
  - #4614 by @satorg
  - #4619 by @i-am-the-slime
  - #4622 by @bwignall
  - #4623 by @danicheg
  - #4624 by @bwignall
  - #4625 by @igstan
  - #4632 by @yanns
  - #4634 by @bwignall
  - #4638 by @danicheg
  - #4641 by @danicheg
  - #4644 by @danicheg
  - #4645 by @danicheg
  - #4648 by @Masynchin
  - #4656 by @johnynek
  - #4659 by @satorg
  - #4660 by @valencik
  - #4662 by @m50d
  - #4667 by @mzuehlke
  - #4670 by @aluscent
  - #4673 by @xuwei-k
  - #4676 by @joroKr21
  - #4678 by @LaurenceWarne
  - #4679 by @bio-aeon
  - #4682 by @satorg
  - #4693 by @durban
  - #4695 by @johnynek

</details>

### Updates

All updates by @typelevel-steward.

<details>

  - #4607
  - #4609
  - #4627
  - #4646
  - #4675
  - #4643
  - #4653
  - #4684
  - #4697
  - #4633
  - #4655
  - #4666
  - #4669
  - #4672
  - #4683
  - #4691
  - #4671
  - #4674
  - #4668
  - #4688
  - #4685
  - #4636
  - #4651
  - #4665
  - #4692
  - #4649
  - #4657
  - #4658
  - #4654
  - #4620
  - #4698

</details>

## Approved

Reviewed PRs that have got at least 1 approval from the organization members.

### Ready to be merged

These PRs have no conflicts and expected to be merged soon unless concerns or objections emerge. Additional reviews and approvals are very welcome but not required for these PRs to get them merged.

  - #4694 by @johnynek (approved by @satorg)

### Approved, but got conflicts

These PRs have got some conflicts due to recent merges. Therefore they can be merged once the conflicts are resolved.

  - #4677 by @satorg (approved by @johnynek)

## Ready for Review

PRs that look pretty complete but either haven't been reviewed or approved, or may need some additional work on the comments to get their approvals.

  - #4564 by @massimosiani (reviewed by @armanbilge, @satorg)

**Comment by satorg** [comment]
There are a couple of PRs in the list (#4528 and #4540) that could be either merged easily or rejected. However I guess, the question on whether to merge or not to merge is more about policies rather than the code itself, isn't it? Are there any formal routines on accepting or rejecting PRs like those two? cc @samspills

**Reply by armanbilge** [reply]
My opinion is we should no longer accept those sorts of PRs in Cats or other projects' websites. Instead, they should apply to become affiliates and become listed on Typelevel.org. Then, there is a definitive list of projects in our ecosystem that are appropriately OSS licensed and have a Code of Conduct.

**Reply by armanbilge** [reply]
Oh, I missed that one is an "adopter" rather than a project. We don't currently have it, but I also think those should all just be centralized somewhere on the Typelevel website instead of repeated project-per-project documentation site ...

**Reply by samspills** [reply]
(I am travelling + doing holiday obligations and don't want to block anything, so I will tag in @typelevel/steering for other folks to chime in!)

**Reply by jducoeur** [reply]
Yeah, the page for #4528 isn't about projects, it's about companies: just a plain "who uses Cats?"

Their website is a bit under-baked -- it looks like a small company still establishing itself -- but I don't see anything unusual or out of bounds about the PR.  I'd probably recommend accepting it.  (But I'm not active as a maintainer on Cats, just mainly focused on policy, so I'll leave it to maintainers to merge it.)

I suspect Arman is right that the adoption page should be centralized, decoupled from Cats per se.  And I'm not 100% sure whether this sort of page is as good an idea as it used to be: it's potentially an SEO vector, and I suspect we'll need to keep a careful eye on PRs in the age of increasingly automated abuse.

But this looks like a legit usage as things stand: what I see from Blue Insight's webpage does appear to be somewhat TL-oriented.

After New Year's, we should chat about moving this page (and others like it, at least for the Organization projects) to a central location somewhere on typelevel.org.

**Reply by djspiewak** [reply]
I vote for merging this one as-is and working to centralize later. Better to decouple.

**Comment by joroKr21** [comment]
Regarding #4303 - I think just nobody reviewed it

**Comment by satorg** [comment]
FYI: I've updated the description with the most recent changes to the `main` branch.

I wonder if'd be possible to automate this process, i.e. make the description auto-generated by a script like I use ([generate-prerelease-summary.sh](https://gist.github.com/satorg/9636a084aec855f21543175e4c361196)) 🤔 
And with all the community brains put together we could make the script even better I guess.

**Reply by mzuehlke** [reply]
Maybe https://github.com/release-drafter/release-drafter could be used for this. But I have till now used it only to generate release notes based labeled pull requests.

**Comment by satorg** [comment]
Turns out that #4678 got conflicts and cannot be merged even though it is approved. I've updated the script to keep PRs like that one tracked and updated the description here after that 🤷

**Reply by kubukoz** [reply]
looks like that PR is mergeable again :)

**Comment by satorg** [comment]
Would we like to assign a specific date for the release? So that everything that will have been approved by that date gets merged and therefore released automatically (unless any concerns emerge by that date), wdyt?

We can schedule it tentatively on, let's say, Jan 1st 😅  but any other date would work too, of course.

**Comment by satorg** [comment]
FYI: updated the discussion board again.

**Comment by satorg** [comment]
Hello there, FYI:
1. I've updated the discussion board to the most recent state.
2. I'm going to merge #4670 in a couple of days. It looks pretty safe and straightforward, but if anyone happens to have any concerns, please speak up.
3. I'm not that confident regarding #4677 though – the PR has been approved by @johnynek (thank you a lot!), but I feel there's still some uncertainty regarding it among the members, so I'm totally fine to put it on hold for the upcoming release. Again, if anyone has an opinion (no matter for or against it), please speak up.

Thanks everyone for your help!

**Reply by kubukoz** [reply]
#4677 also seems like something that doesn't necessarily need to happen in a minor release, given it's imports inside the codebase (and not the imports our users will write), I'd say it can be dealt with at a later time

**Reply by satorg** [reply]
Yeah, anyhow, I'd prefer to be sure that everyone feels pretty comfortable with that change. No reason to push it through.

**Comment by satorg** [comment]
Hi everyone! I believe we did a pretty god job on cleaning pending PRs up, so that now `main` looks good to go live, doesn't it?
Wdyt guys, should we kick the new release off?

**Reply by kubukoz** [reply]
I think it's time :)

**Comment by satorg** [comment]
Hello everyone!

I've created a draft for the v2.13.0 release (as GitHub assured me, collaborators should be able to see it in the [releases](https://github.com/typelevel/cats/releases) section).

Before I hit "Publish", could everyone who cares look through the draft and let me know if there are any discrepancies in the description please? Or go ahead and fix them yourselves with no doubt! The reason I'm asking is that initially GitHub generated a description where most of the tickets fell into "Uncategorized" section with a few exceptions. I managed to categorize them, but I might have overlooked something, so I'd appreciate your help with the verification. Thank you!

<sub>
cc @kubukoz @armanbilge @danicheg @djspiewak @mzuehlke @joroKr21 @johnynek @jducoeur (and everyone else, sorry if I forgot to cc you)
</sub>

**Reply by kubukoz** [reply]
sorry for not responding, I forgot to comment that the draft wasn't visible to me at the time.

**Reply by satorg** [reply]
Yeah, I'm not sure how it works... I asked Github Copilot, it assured me that collaborators should see release drafts :) But perhaps it meant something else or was hallucinating again.

**Reply by kubukoz** [reply]
see, I'm not a collaborator, that'd explain it ;)

**Reply by joroKr21** [reply]
Same for me 😄

**Comment by satorg** [comment]
[v2.13.0](https://github.com/typelevel/cats/releases/tag/v2.13.0) is published!

**Reply by kubukoz** [reply]
https://github.com/typelevel/cats/actions/runs/12874302171/job/35893905153

looks like the tests are randomly failing, perhaps another retry will do it? 😅

**Reply by satorg** [reply]
scala-native thing again :(

**Comment by satorg** [comment]
v2.13.0 hit Maven Central 🎉
